plotille/_figure.py

```diff
@@ -238,17 +239,56 @@ def _x_axis(self, xmin, xmax, label='X', with_y_axis=False):
     def clear(self):
         self._plots = []
 
+    def circle(self, xcenter=2.5, ycenter=5,  radius=10, label=None, interp='linear', lc=None):
+        # circle is just a special case of ellipse
+        self.ellipse(xcenter=xcenter, ycenter=ycenter, angle=0,
+                     xamplitude=radius, yamplitude=radius,
+                     label=label, interp=interp, lc=lc)
+
+    def ellipse(self, xcenter=2.5, ycenter=5, angle=30,
+                xamplitude=1,  yamplitude=2.5,
+                label=None, interp='linear', lc=None):
+
+        # x-position of the center
+        u = xcenter
+        # y-position of the center
+        v = ycenter
+        # radius on the x-axis
+        a = xamplitude
+        # radius on the y-axis
+        b = yamplitude
+        # rotation angle
+        t_rot = angle
+
+        t = np.linspace(0, 2*np.pi, 100)
+        ell = np.array([a*np.cos(t), b*np.sin(t)])
+        r_rot = np.array([[np.cos(t_rot), -np.sin(t_rot)], [np.sin(t_rot), np.cos(t_rot)]])
+        ell_rot = np.zeros((2, ell.shape[1]))
+        for i in range(ell.shape[1]):
+            ell_rot[:, i] = np.dot(r_rot, ell[:, i])
+
+        X = u+ell_rot[0, :]  # noqa: N806
+        Y = v+ell_rot[1, :]  # noqa: N806
+        self.plot(X, Y, lc=lc, label=label, interp='linear')  # noqa: N803
+
     def plot(self, X, Y, lc=None, interp='linear', label=None):  # noqa: N803
         if len(X) > 0:
             if lc is None:
                 lc = next(self._color_seq)[self.color_mode]
-            self._plots += [Plot.create(X, Y, lc, interp, label)]
+            overlay = False
+            self._plots += [Plot.create(X, Y, lc, interp, label, overlay)]
 
-    def scatter(self, X, Y, lc=None, label=None):  # noqa: N803
+    def scatter(self, X, Y, lc=None, label=None, marker='', text=None):  # noqa: N803
```

Comment:
Better use `None` to indicate the field is not used?

```suggestion
    def scatter(self, X, Y, lc=None, label=None, marker=None, text=None):  # noqa: N803
```

The test later can be something like: `if marker:`